### PR TITLE
Trusted Advisor integration tests

### PR DESCRIPTION
**Pull request by nico-stefani**
|Related issue|
|-------------|
|      #3341     |

## Description

This PR adds tests for AWS Trusted Advisor integration

#### Tier 0
- Default configuration tests
- `remove_from_bucket` parameter tests
- `only_logs_after` parameter tests
- `path` parameter tests
- `discard_regex` parameter tests

#### Tier 1
- `only_logs_after` parameter tests
---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @nico-stefani  (Developer)  | `tests/integration/test_aws` | ⚫⚫⚫ | :green_circle: :red_circle:  | Ubuntu 22.04    |  [^1]  | Nothing to highlight |
| @reviewer (Reviewer)   | | ⚫⚫⚫ |  |   |      | Nothing to highlight |

### Tier 0 results
```python
root@ubuntu-jammy:/home/vagrant/qa/tests/integration/test_aws# pytest -k trusted --tier 0
================================================================================================ test session starts ================================================================================================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/vagrant/qa/tests/integration, configfile: pytest.ini
plugins: metadata-2.0.2, html-3.1.1, testinfra-5.0.0
collected 24 items / 16 deselected / 8 selected

test_basic.py .                                                                                                                                                                                               [ 12%]
test_discard_regex.py .                                                                                                                                                                                       [ 25%]
test_only_logs_after.py ..                                                                                                                                                                                    [ 50%]
test_path.py ...                                                                                                                                                                                              [ 87%]
test_remove_from_bucket.py .                                                                                                                                                                                  [100%]

================================================================================================= warnings summary ==================================================================================================
../../../../../../usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:433
  /usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:433: PytestCacheWarning: cache could not write path /home/vagrant/qa/tests/integration/.pytest_cache/v/cache/nodeids
    config.cache.set("cache/nodeids", sorted(self.cached_nodeids))

../../../../../../usr/local/lib/python3.10/dist-packages/_pytest/stepwise.py:52
  /usr/local/lib/python3.10/dist-packages/_pytest/stepwise.py:52: PytestCacheWarning: cache could not write path /home/vagrant/qa/tests/integration/.pytest_cache/v/cache/stepwise
    session.config.cache.set(STEPWISE_CACHE_DIR, [])

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================= 8 passed, 16 deselected, 2 warnings in 171.01s (0:02:51) ==============================================================================
```

### Tier 1 results

```python
root@ubuntu-jammy:/home/vagrant/qa/tests/integration/test_aws# pytest -k trusted --tier 1
============================= test session starts ==============================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/vagrant/qa/tests/integration, configfile: pytest.ini
plugins: metadata-2.0.2, html-3.1.1, testinfra-5.0.0
collected 24 items / 23 deselected / 1 selected

test_only_logs_after.py F                                                [100%]

=================================== FAILURES ===================================
------------------------------ Captured log call -------------------------------
DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-trusted-advisor-integration-tests', '--type', 'custom', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2']'
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Table does not exist; create
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: 2023/01/16
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping file with another prefix: test_prefix/2022/11/21/firehose_trustedadvisor-1-20221121T1932Z.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-trusted-advisor-integration-tests', '--type', 'custom', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2', '--only_logs_after', '2022-NOV-20']'
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: 2022/11/20
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Found new log: 2022/11/21/firehose_trustedadvisor-1-20221121T1932Z.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Found new log: 2022/11/23/firehose_trustedadvisor-1-20221123T1932Z.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Found new log: 2022/11/26/firehose_trustedadvisor-1-20221126T1932Z.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping file with another prefix: test_prefix/2022/11/21/firehose_trustedadvisor-1-20221121T1932Z.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-trusted-advisor-integration-tests', '--type', 'custom', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2', '--only_logs_after', '2022-NOV-20']'
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: 2022/11/20
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: 2022/11/21/firehose_trustedadvisor-1-20221121T1932Z.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: 2022/11/23/firehose_trustedadvisor-1-20221123T1932Z.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: 2022/11/26/firehose_trustedadvisor-1-20221126T1932Z.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping file with another prefix: test_prefix/2022/11/21/firehose_trustedadvisor-1-20221121T1932Z.json
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
ERROR    wazuh_testing:cli_utils.py:62 Some logs may were processed
ERROR    wazuh_testing:cli_utils.py:63 Results found: 3
ERROR    wazuh_testing:cli_utils.py:64 Results expected: 1
=============================== warnings summary ===============================
../../../../../../usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:433
  /usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:433: PytestCacheWarning: cache could not write path /home/vagrant/qa/tests/integration/.pytest_cache/v/cache/nodeids
    config.cache.set("cache/nodeids", sorted(self.cached_nodeids))

../../../../../../usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:387
  /usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:387: PytestCacheWarning: cache could not write path /home/vagrant/qa/tests/integration/.pytest_cache/v/cache/lastfailed
    config.cache.set("cache/lastfailed", self.lastfailed)

../../../../../../usr/local/lib/python3.10/dist-packages/_pytest/stepwise.py:52
  /usr/local/lib/python3.10/dist-packages/_pytest/stepwise.py:52: PytestCacheWarning: cache could not write path /home/vagrant/qa/tests/integration/.pytest_cache/v/cache/stepwise
    session.config.cache.set(STEPWISE_CACHE_DIR, [])

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED test_only_logs_after.py::test_multiple_calls[trusted_advisor_only_logs_after_multiple_calls]
================ 1 failed, 23 deselected, 3 warnings in 25.82s =================

```
[^1]: Because the module is showing the message `Skipping file with another prefix: ...` multiple times, the *tier 1* test is falling. This should be fixed in his respective issue.

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py [hunk 1]
```diff
@@ -94,22 +103,204 @@ def get_data_sample(self) -> dict:
...
+            data.append(
+                [
+                    "2", cons.RANDOM_ACCOUNT_ID, get_random_interface_id(), get_random_ip(), get_random_ip(),
+                    get_random_port(), get_random_port(), "6", "39", "4698", "1622505433", "1622505730", "ACCEPT", "OK"
+                ]
+            )
+        buffer = StringIO()
+        csv.writer(buffer, delimiter=" ").writerows(data)
+
+        return buffer.getvalue()
```
Comment (nico-stefani) [inline deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:144]: For the reviewer, please disregard this already has been fixed in https://github.com/wazuh/wazuh-qa/pull/3699/.

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py [hunk 2]
```diff
@@ -94,22 +103,204 @@ def get_data_sample(self) -> dict:
...
+                            'ARN': f"arn:aws:kms:us-east-1:{cons.RANDOM_ACCOUNT_ID}:key/{str(uuid4())}",
+                            'accountId': cons.RANDOM_ACCOUNT_ID,
+                            'type': 'AWS::KMS::Key'
+                        }
+                    ],
+                    'eventType': 'AwsApiCall',
+                    'vpcEndpointId': f"vpce-{get_random_string(17)}"
+                }
+            }
+        )
```
Comment (nico-stefani) [inline deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:228]: For the reviewer, please disregard this already has been fixed in https://github.com/wazuh/wazuh-qa/pull/3715/.

File: tests/integration/test_aws/test_discard_regex.py [hunk 3]
```diff
@@ -128,10 +128,13 @@ def test_discard_regex(
...
     wazuh_log_monitor.start(
-        timeout=10,
-        callback=event_monitor.make_aws_callback(pattern),
-        error_message='The AWS module did not show correct message about discard regex',
-        accum_results=skipped_logs
+        timeout=T_20,
+        callback=event_monitor.callback_detect_event_processed_or_skipped(pattern),
+        error_message=(
+            'The AWS module did not show correct message about discard regex or ',
+            'did not process the expected amout of logs'
```
Comment (fdalmaup) [inline tests/integration/test_aws/test_discard_regex.py:135]:
```suggestion
            'The AWS module did not show the correct message about discard regex or ',
            'did not process the expected amount of logs'
```

**Review by fdalmaup — CHANGES_REQUESTED** [review]

**Review by fdalmaup — APPROVED** [review]

**Comment by fdalmaup** [comment]
LGTM

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**